File: common/src/main/java/com/hartwig/actin/molecular/datamodel/pharmaco/PharmacoEntry.java
package com.hartwig.actin.molecular.datamodel.pharmaco;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PharmacoEntry {

    @NotNull
    public abstract String gene();

    @NotNull
    public abstract String result();
}

File: common/src/main/kotlin/com/hartwig/actin/molecular/datamodel/orange/pharmaco/PharmacoEntry.kt
package com.hartwig.actin.molecular.datamodel.orange.pharmaco

data class PharmacoEntry(
    val gene: PharmacoGene,
    val haplotypes: Set<Haplotype>
) : Comparable<PharmacoEntry> {

    override fun compareTo(other: PharmacoEntry): Int {
        return Comparator.comparing(PharmacoEntry::gene)
            .thenComparing({ it.haplotypes.size }, Int::compareTo)
            .compare(this, other)
    }
}
